### PR TITLE
[BUGFIX] Drop native return types for `getAccessibleMock` & friends

### PR DESCRIPTION
**Pull request by oliverklee**
The combination of "union types with generics in PHPDoc" and
"a subset of those types as native types, but without the generics"
tends to confuse PhpStorm's static type analysis (which we want to avoid).

So we stay with PHPDoc return type annotations in the affected places.

Releases: main

### CODE REVIEW

File: Classes/Core/BaseTestCase.php [hunk 1]
```diff
@@ -30,6 +30,9 @@ abstract class BaseTestCase extends TestCase
     /**
      * Creates a mock object which allows for calling protected methods and access of protected properties.
      *
+     * Note: This method has no native return types as having native (incomplete) return types, PHPDoc return type
+     * annotations and generics in combination tends to confuse PhpStorm's static type analysis.
```
Comment (sbuerk) [inline Classes/Core/BaseTestCase.php:34]: Mhh .. this sentens sounds kind weired. "has no native return typas as having native (inclomplete) return types..." and so go ...

File: Classes/Core/BaseTestCase.php [hunk 2]
```diff
@@ -81,6 +84,9 @@ protected function getAccessibleMock(
      * of protected properties. Concrete methods to mock can be specified with
      * the last parameter
      *
+     * Note: This method has no native return types as having native (incomplete) return types, PHPDoc return type
```
Comment (sbuerk) [inline Classes/Core/BaseTestCase.php:87]: same as above - sounds weired.

**Review by sbuerk — CHANGES_REQUESTED** [review]
May you be up to rephrase the two (same) comments ? Otherwise change/PR looks good and works so far.

**Comment by lolli42** [comment]
how to continue here? if monorepo-main is fine with this, i'd tend to be fine with it, too? maybe rephrase wording a bit?

**Comment by oliverklee** [comment]
Yes, I‘ll update the wording (probably on Saturday, i.e., tomorrow).

File: Classes/Core/BaseTestCase.php (hunk 1 above)
Comment (oliverklee) [reply Classes/Core/BaseTestCase.php]: I have rephrased those two comments now. I hope that they now are less weird and less confusing. :-)
